xdsl/dialects/builtin.py

```diff
@@ -339,7 +339,9 @@ class FloatData(Data[float]):
 
     @staticmethod
     def parse_parameter(parser: BaseParser) -> float:
-        return parser.parse_float_literal()
+        span = parser.expect(parser.try_parse_float_literal,
+                             "Expect float literal")
```

Comment:
Maybe a little bit more context on the error message? Something like `Malformed Float data: Expected a float literal!`

Reply:
Yeah here also, I'm not sure that it adds any information